packages/components/src/color-picker/saturation.js

```diff
@@ -0,0 +1,151 @@
+		this.mousetrap.bind( 'pageup', () => this.brighten( 1 ) );
+		this.mousetrap.bind( 'down', () => this.brighten( -0.01 ) );
+		this.mousetrap.bind( 'shift+down', () => this.brighten( -0.1 ) );
+		this.mousetrap.bind( 'pagedown', () => this.brighten( -1 ) );
+
+		this.mousetrap.bind( 'right', () => this.saturate() );
+		this.mousetrap.bind( 'shift+right', () => this.saturate( 0.1 ) );
+		this.mousetrap.bind( 'end', () => this.saturate( 1 ) );
+		this.mousetrap.bind( 'left', () => this.saturate( -0.01 ) );
+		this.mousetrap.bind( 'shift+left', () => this.saturate( -0.1 ) );
+		this.mousetrap.bind( 'home', () => this.saturate( -1 ) );
+	}
+
+	componentWillUnmount() {
+		this.throttle.cancel();
+		this.unbindEventListeners();
+		this.mousetrap.reset();
+	}
+
+	saturate( amount = 0.01 ) {
+		const { hsv, onChange = noop } = this.props;
+		const intSaturation = clamp(
+			Math.round( hsv.s * 100 ) + Math.round( amount * 100 ),
+			0,
+			100
+		);
+		const change = {
+			h: hsv.h,
+			s: intSaturation / 100,
+			v: hsv.v,
+			a: hsv.a,
+			source: 'rgb',
+		};
+
+		onChange( change );
+	}
+
+	brighten( amount = 0.01 ) {
+		const { hsv, onChange = noop } = this.props;
+		const intValue = clamp(
+			Math.round( hsv.v * 100 ) + Math.round( amount * 100 ),
+			0,
+			100
+		);
+		const change = {
+			h: hsv.h,
+			s: hsv.s,
+			v: intValue / 100,
+			a: hsv.a,
+			source: 'rgb',
+		};
+
+		onChange( change );
+	}
+
+	handleChange( e ) {
+		const { onChange = noop } = this.props;
+		const change = calculateSaturationChange( e, this.props, this.container );
+		this.throttle( onChange, change, e );
+	}
+
+	handleMouseDown( e ) {
+		this.handleChange( e );
+		window.addEventListener( 'mousemove', this.handleChange );
+		window.addEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	handleMouseUp() {
+		this.unbindEventListeners();
+	}
+
+	unbindEventListeners() {
+		window.removeEventListener( 'mousemove', this.handleChange );
+		window.removeEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	render() {
+		const { hsv, hsl, instanceId } = this.props;
+		const pointerLocation = {
+			top: `${ -( hsv.v * 100 ) + 100 }%`,
+			left: `${ hsv.s * 100 }%`,
+		};
+
+		/* eslint-disable jsx-a11y/no-static-element-interactions */
+		return (
+			<div
+				style={ { background: `hsl(${ hsl.h },100%, 50%)` } }
+				className="components-color-picker__saturation-color"
+				ref={ ( container ) => ( this.container = container ) }
+				onMouseDown={ this.handleMouseDown }
+				onTouchMove={ this.handleChange }
+				onTouchStart={ this.handleChange }>
+				<div className="components-color-picker__saturation-white" />
+				<div className="components-color-picker__saturation-black" />
+				<button
+					aria-label={ __( 'Saturation & lightness value' ) }
+					aria-describedby={ `color-picker-saturation-${ instanceId }` }
+					className="components-color-picker__saturation-pointer"
+					style={ pointerLocation }
+				/>
```

Comment:
Initial focus goes to this button and it gets announced as:
"Saturation & lightness value button"

![screenshot 104](https://user-images.githubusercontent.com/1682452/47096275-1efd1b00-d22f-11e8-92e0-f312f28056df.png)

I'm not sure a `button` is the best element to use, because its role suggests it can be pressed, not "moved" on a canvas. In combination with its label `Saturation & lightness value button` I'm not sure it's clear enough what this control is about for users. How this could be improved? I can't think of an ARIA role that is suitable for this control. However, maybe we can try to improve its aria-label. Considering initial focus is set on this button, instead of 

`Saturation & lightness value`

I'd propose to simplify and use:

`Pick a color` - followed by the detailed aria-description - `Use your arrow keys to change ...` 

A combination of label and description along those lines could help.

Reply:
"Color" might be confusing here, since this area is for tweaking the saturation and lightness, and there's a second slider below that allow for selecting more of what people think of as "colour" (the hue, that is). 

It's not *technically* correct, per se, but I think we could get away with saying "Choose a shade" here and have people understand generally what that refers to.

Reply:
I've updated the label to read "Choose a shade" in 51f12019004992a3f2618880e3bb773fd75ceb93 . I haven't moved the components though, I think for now we should leave the sliders as they are (and maybe follow up with another PR if it still feels too awkward).